core/variant/type_info.h

```diff
@@ -367,4 +367,82 @@ ZERO_INITIALIZER_NUMBER(char32_t)
 ZERO_INITIALIZER_NUMBER(float)
 ZERO_INITIALIZER_NUMBER(double)
 
+template <typename T>
+inline constexpr Variant::Type get_variant_type_v = TypeInherits<Object, T>::value ? Variant::OBJECT : Variant::VARIANT_MAX;
+
+template <typename T>
+inline constexpr bool is_variant_type_v = get_variant_type_v<T> != Variant::VARIANT_MAX;
+
+template <typename T>
+struct VariantComplexConstants {
```

Comment:
The name of this struct confuses me. Beginning with `Variant` makes me think it belongs in one of the `variant*.h` files. What is `Complex` meant to mean here? Maybe call the struct something like `GetClassName` and call the methods `self` and `parent` so you use it like `GetClassName<T>::self()` or `GetClassName<T>::parent()`.

Reply:
I had the broader name for two reasons: in case I wanted to revisit this in the future (that was silly in hindsight), and to convey that this is a Variant-specific context. The name/functions you proposed are much nicer though, and a similar exclusion effect could be accomplished with a static assertion similar to `TypedArray`, so I'll change to that instead.

Reply:
Went with `GetObjectClassName` instead, as `GetClassName` is already defined in `winuser.h` & this makes the object-specific context more obvious.